### PR TITLE
Implementation of the purchasing function by payjp

### PR DESCRIPTION
**Pull request by MasatakaOba**
## WHAT
Add Implementation of the purchasing function.

The function details are below. 
1. Credit card registration/delation function.
2. Buying items function

These functions is in cooperation with Payjp database.
I have already conform that  cooperation is normally performed.

notice: Item database has not been created yet. So shortly after created, I will link database to this function.

## WHY
To be able to purchase items.

### Credit card registration/delation
f25b48e702231a9af6b1f253f080f25f

### Purchasing an item page image
1f489e069dc11c86a3b67a77bd7b74ec

### CODE REVIEW

File: app/controllers/purchase_controller.rb [hunk 1]
```diff
@@ -1,2 +1,30 @@
 class PurchaseController < ApplicationController
-end
+
+  def index
+    card = Card.where(user_id: current_user.id).first
+    if card.blank?
+      redirect_to controller: "card", action: "new"
+    else
+      require 'payjp'
```
Comment (yuki-uchida) [inline app/controllers/purchase_controller.rb]: require 'payjp'は2行目にまとめても良いのでは？

**Review by yuki-uchida — CHANGES_REQUESTED** [review]

File: app/controllers/purchase_controller.rb (hunk 1 above)
Comment (MasatakaOba) [reply app/controllers/purchase_controller.rb]: クラスの呼び出しごとにrequire 'payjp'を呼び出してしまって動作が重くなってしまうと考え分けておりました。
こちら記述を2行目に統一しました。

**Comment by yuki-uchida** [comment]
LGTMです

**Comment by MasatakaOba** [comment]
> LGTMです

ありがとうございます！

**Pull request merged**